adr: 0036 Run ingress-nginx as a daemon set (#538)
elastisys/compliantkubernetes: docs/adr/0036-run-ingress-nginx-as-daemonset.md

> # Run ingress-nginx as a daemon set

Reviewer: I think the proper kubernetes term should be used
```suggestion
# Run ingress-nginx as a daemonSet
```

Author: I agree
Changed now
Thx

* Status: accepted
* Deciders: arch meeting
* Date: 2023-03-16

## Context and Problem Statement

> Currently we run the ingress-nginx as a daemon set. This  feels like a waste of resources on environments with many nodes, and we can instead run it as a deplyment and have only 2 replicas running.

Reviewer: ```suggestion
Currently we run the ingress controller, `ingress-nginx`, as a daemonSet.
This can potentially feels like a waste of resources on worker nodes in large environments.
Running the ingress controller as a deployment with at least two replicas is a possibility.
```

Author: Changed

Should we run ingress-nginx as a deployment or as a daemon set?
What do we do for Infra Providers that do not have service type loadbalancer?

## Decision Drivers

* We want to deliver a stable and secure platform.
* Want want to keep application Nodes "light", so the application team knows what capacity is available for their application.
* We want to find a solution which is scalable and minimizes administrator burden.
* We don't want to waste infrastructure.
* We want to keep things simple.

## Considered Options

> * Keep running the ingress-nginx as a daemon set.
> * Run ingress-nginx as a deployment with 2 or more replicas depending on the nevironment size and requirements.
> * Do not run ingress-nginx on the AMS nodes.
> * For Infra Providers without service type loadbalancer continue using host network as decided in adr0008
> * For Infra Providers without service type loadbalancer start using service type NodePort for nginx and also use the external load balancer to route traffic from ports 80/443 to node ports 30080/30443.

Reviewer: Could we use an ordered list instead?
```suggestion
1. Keep running the ingress-nginx as a daemon set.
1. Run ingress-nginx as a deployment with 2 or more replicas depending on the nevironment size and requirements.
1. Do not run ingress-nginx on the AMS nodes.
1. For Infra Providers without service type loadbalancer continue using host network as decided in adr0008
1. For Infra Providers without service type loadbalancer start using service type NodePort for nginx and also use the external load balancer to route traffic from ports 80/443 to node ports 30080/30443.
```

# Run ingress-nginx as a daemon set

* Status: accepted
* Deciders: arch meeting
* Date: 2023-03-16

## Context and Problem Statement

> Currently we run the ingress-nginx as a daemon set.

Reviewer: ```suggestion
Currently we run ingress-nginx as a daemon set.
```

> This can potentiolly feel like a waste of resources in large environments.

Reviewer: ```suggestion
This can potentially feel like a waste of resources in large environments.
```

Author: Fixed
Thx

> Running the ingress controller as a deployment with at least two replicas is a possibility.

Reviewer 2: Nit-pick, new paragraph for the problem statement.

```suggestion
Running the ingress controller as a deployment with at least two replicas is a possibility.

```

Author: Fixed now
Thx

Should we run ingress-nginx as a deployment or as a daemonSet?
What do we do for Infra Providers that do not have service type loadbalancer?

## Decision Drivers

* We want to deliver a stable and secure platform.
* Want want to keep application Nodes "light", so the application team knows what capacity is available for their application.
* We want to find a solution which is scalable and minimizes administrator burden.
* We don't want to waste infrastructure.
* We want to keep things simple.

## Considered Options

1. Keep running the ingress-nginx as a daemonSet.
1. Run ingress-nginx as a deployment with 2 or more replicas depending on the environment size and requirements.
1. Do not run ingress-nginx on the AMS nodes.
1. For Infra Providers without service type loadbalancer continue using host network as decided in adr0008
1. For Infra Providers without service type loadbalancer start using service type NodePort for nginx and also use the external load balancer to route traffic from ports 80/443 to node ports 30080/30443.

## Decision Outcome

Chosen options: 1 & 3 & 5
1. "Keep running ingress-nginx as a daemonSet."
3. "Do not run ingress-nginx on the AMS nodes."

> 5. "For Infra Providers without service type loadbalancer start using service type NodePort for nginx and also use the external load balancer to route traffic from ports 80/443 to node ports 30080/30443" -> This superseeds adr0008.

Author: Yeah,
I don't know how to enforce it to produce a list of 1,3,5 ... Do you know how?
Or should I just do 1. 1. 1. and have it produce the default  1.  2.  3.  list?

> 5. "For Infra Providers without service type loadbalancer start using service type NodePort for nginx and also use the external load balancer to route traffic from ports 80/443 to node ports 30080/30443" -> This superseeds adr0008.

Author: Replaced with *
Thx

### Positive Consequences

* We keep things simple and have the same solution on all Infrastructure Providers.

> * We keep the platform stable and secure and not risc when we replace nodes or nodes become unavailable.

Reviewer 3: ```suggestion
* We keep the platform stable and secure and not risk when we replace nodes or nodes become unavailable.
```

Author: fixed
